ui/packages/catalog-ui-search/src/main/webapp/react-component/presentation/result-sort/result-sort.tsx

```diff
@@ -0,0 +1,137 @@
+/**
```

Comment:
I think you want to move the state logic into a container component at webapp/react-component/container, then pass the relevant functions into this component as props.

Reply:
done